File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.String/String.ExtractLetter.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.Linq

Public Module Extensions_495

	''' <summary>
	'''     A string extension method that extracts the letter described by @this.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <returns>The extracted letter.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function ExtractLetter(this As String) As String
		Return New String(this.ToCharArray().Where(Function(x) [Char].IsLetter(x)).ToArray())
	End Function
End Module

File: src/Z.Core/System.String/String.ExtractLetter.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
using System;
using System.Linq;

public static partial class Extensions
{
    /// <summary>
    ///     A string extension method that extracts the letter described by @this.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <returns>The extracted letter.</returns>
    public static string ExtractLetter(this string @this)
    {
        return new string(@this.ToCharArray().Where(x => Char.IsLetter(x)).ToArray());
    }
}